python/python/lance/dataset.py

```diff
@@ -689,6 +689,83 @@ def cleanup_old_versions(
             td_to_micros(older_than), delete_unverified
         )
 
+    def create_scalar_index(
+        self,
+        column: Union[str, List[str]],
```

Comment:
In Python, it wouldn't be a breaking API change to go from `str` to `Union[str, List[str]]`, so since we don't yet support multiple columns why don't we just keep this as `str` for now?

Reply:
I've updated this to `str`.